Bukkit/src/main/java/com/plotsquared/bukkit/listener/BlockEventListener.java

```diff
@@ -1183,4 +1184,37 @@ public void onLeavesDecay(LeavesDecayEvent event) {
 
     }
 
+    @EventHandler(ignoreCancelled = true)
+    public void onSpongeAbsorb(SpongeAbsorbEvent event) {
+        Block sponge = event.getBlock();
+        Location location = BukkitUtil.adapt(sponge.getLocation());
+        PlotArea area = location.getPlotArea();
+        List<org.bukkit.block.BlockState> blocks = event.getBlocks();
+        if (area == null) {
+            for (int i = blocks.size() - 1; i >= 0; i--) {
```

Comment:
Is there any reason why the list is iterated backwards? If not, a simple `removeIf` would be better probably (and if it needs to be reversed, we could probably use the `ListIterator` instead).

Reply:
No particular reason. I copied this loop from https://github.com/IntellectualSites/PlotSquared/blob/6e7bd0a5365040f9579ff369f11761fd5834c5b3/Bukkit/src/main/java/com/plotsquared/bukkit/listener/BlockEventListener.java#L982-L987
Similarly for the backwards loop below. Changing it to `removeIf` sounds good to me.

Reply:
Yeah I guess you can change it then.